javac looking in sdkman which is not currently there

```
jbang --verbose https://github.com/jhy/jsoup/blob/master/src/main/java/org/jsoup/examples/ListLinks.java
JAVA_HOME is set but does not seem to point to a valid Java JDK
[jbang] jbang version 0.83.1
[jbang] https://github.com/jhy/jsoup/blob/master/src/main/java/org/jsoup/examples/ListLinks.java is not from a trusted source thus not running it automatically.

If you trust the url to be safe to run you can do one of the following:
0) Trust once: Add no trust, just run this time
1) Trust limited url in future:
    jbang trust add https://github.com/jhy/jsoup/


Any other response will result in exit.

[jbang] Type in your choice (0 or 1) and hit enter. Times out after 10 seconds.
0
[jbang] Retrieved file from cache https://raw.githubusercontent.com/jhy/jsoup/master/src/main/java/org/jsoup/examples/ListLinks.java = /Users/max/.jbang/cache/urls/5b67081dc7609bec867dc9ddca86d74daa64aa60486ce9212e1d1772698686a0/ListLinks.java
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] Building jar...
[jbang] compile: /Users/max/.sdkman/candidates/java/current/bin/javac -d /Users/max/.jbang/cache/jars/ListLinks.java.0186ce57dfab252beee2b196b86c94c320abd4b2c1eb1fb7ae8982223c15c594.jar.tmp /Users/max/.jbang/cache/urls/5b67081dc7609bec867dc9ddca86d74daa64aa60486ce9212e1d1772698686a0/ListLinks.java
[jbang] Deleting folder /Users/max/.jbang/cache/jars/ListLinks.java.0186ce57dfab252beee2b196b86c94c320abd4b2c1eb1fb7ae8982223c15c594.jar.tmp
[jbang] [ERROR] Cannot run program "/Users/max/.sdkman/candidates/java/current/bin/javac": error=2, No such file or directory
java.io.IOException: Cannot run program "/Users/max/.sdkman/candidates/java/current/bin/javac": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at dev.jbang.cli.BaseBuildCommand.buildJar(BaseBuildCommand.java:168)
	at dev.jbang.cli.BaseBuildCommand.build(BaseBuildCommand.java:122)
	at dev.jbang.cli.BaseBuildCommand.buildIfNeeded(BaseBuildCommand.java:86)
	at dev.jbang.cli.Run.prepareArtifacts(Run.java:99)
	at dev.jbang.cli.Run.doCall(Run.java:76)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:85)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:12)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:135)
	at dev.jbang.cli.JBang$3.handle(JBang.java:130)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 17 more
[jbang] If you believe this a bug in jbang open issue at https://github.com/jbangdev/jbang/issues
```

javac is in PATH:
```
which javac
/usr/bin/javac
```

--- a/src/main/scripts/jbang
+++ b/src/main/scripts/jbang
@@ -133,6 +133,7 @@ fi
 if [[ -z "$JAVA_EXEC" ]]; then
   # Determine if a (working) JDK is available on the PATH
   if [ -x "$(command -v javac)" ]; then
+    unset JAVA_HOME
     JAVA_EXEC="java";
   elif [ -x "$JBDIR/currentjdk/bin/javac" ]; then
     export JAVA_HOME="$JBDIR/currentjdk"

--- a/src/main/scripts/jbang.cmd
+++ b/src/main/scripts/jbang.cmd
@@ -40,6 +40,7 @@ if "!JAVA_EXEC!"=="" (
   rem Determine if a (working) JDK is available on the PATH
   where javac > nul 2>&1
   if !errorlevel! equ 0 (
+    set JAVA_HOME=
     set JAVA_EXEC=java.exe
   ) else if exist "%JBDIR%\currentjdk\bin\javac" (
     set JAVA_HOME=%JBDIR%\currentjdk

--- a/src/main/scripts/jbang.ps1
+++ b/src/main/scripts/jbang.ps1
@@ -106,6 +106,7 @@ if ($JAVA_EXEC -eq "") {
   # Determine if a (working) JDK is available on the PATH
   $ok=$false; try { if (Get-Command "javac") { $ok=$true } } catch {}
   if ($ok) {
+    $env:JAVA_HOME=""
     $JAVA_EXEC="java.exe"
   } elseif (Test-Path "$JBDIR\currentjdk\bin\javac") {
     $env:JAVA_HOME="$JBDIR\currentjdk"

